adoc/headers/vec.h

```diff
@@ -121,112 +121,65 @@ template <typename DataT, int NumElements> class vec {
+
+  // OP is: <<=, >>=
+  /* Available only when: DataT != float && DataT != double && DataT != half. */
+  friend vec& operatorOP(vec& lhs, const vec& rhs);
+  friend vec& operatorOP(vec& lhs, const DataT& rhs);
 
   // OP is: &&, ||
-  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs) {
-    /* ... */ }
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs) {
-    /* ... */ }
-
-    // OP is: <<, >>
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const vec& lhs, const vec& rhs) { /* ... */
-    }
-    friend vec operatorOP(const vec& lhs, const DataT& rhs) { /* ... */
-    }
-
-    // OP is: <<=, >>=
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec& operatorOP(vec& lhs, const vec& rhs) { /* ... */
-    }
-    friend vec& operatorOP(vec& lhs, const DataT& rhs) { /* ... */
-    }
-
-    // OP is: ==, !=, <, >, <=, >=
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs) {
-    /* ... */ }
-    friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs) {
-    /* ... */ }
-
-    vec& operator=(const vec<DataT, NumElements>& rhs);
-    vec& operator=(const DataT& rhs);
-
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operator~(const vec& v) { /* ... */
-    }
-    friend vec<RET, NumElements> operator!(const vec& v) { /* ... */
-    }
-
-    // OP is: +, -, *, /, %
-    /* operator% is only available when: DataT != float && DataT != double &&
-    DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: &, |, ^
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: &&, ||
-    friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs) {
-    /* ... */ }
-
-    // OP is: <<, >>
-    /* Available only when: DataT != float && DataT != double
-    && DataT != half. */
-    friend vec operatorOP(const DataT& lhs, const vec& rhs) { /* ... */
-    }
-
-    // OP is: ==, !=, <, >, <=, >=
-    friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs) {
-    /* ... */ }
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs);
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs);
+  friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs);
+
+  // OP is: ==, !=, <, >, <=, >=
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const vec& rhs);
+  friend vec<RET, NumElements> operatorOP(const vec& lhs, const DataT& rhs);
+  friend vec<RET, NumElements> operatorOP(const DataT& lhs, const vec& rhs);
+
+  vec& operator=(const vec<DataT, NumElements>& rhs);
+  vec& operator=(const DataT& rhs);
```

Comment:
Let's move these assignment operators, so they appear right after the subscript operators.  This will match the order they have in the table.

Reply:
Done in c42cf1a23905537df233b96500914040b772b1ce